src/DotVVM.Framework.Tests.Common/Runtime/ConfigurationSerializationTests.cs

```diff
@@ -0,0 +1,150 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using CheckTestOutput;
+using DotVVM.Framework.Compilation;
+using DotVVM.Framework.Configuration;
+using DotVVM.Framework.Controls;
+using DotVVM.Framework.Hosting;
+using DotVVM.Framework.ResourceManagement;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace DotVVM.Framework.Tests.Runtime
+{
+    [TestClass]
+    public class ConfigurationSerializationTests
+    {
+        OutputChecker check = new OutputChecker("config-tests");
+
+        void checkConfig(DotvvmConfiguration config, string checkName = null, string fileExtension = "json", [CallerMemberName] string memberName = null, [CallerFilePath] string sourceFilePath = null)
+        {
+            var serialized = DotVVM.Framework.Hosting.VisualStudioHelper.SerializeConfig(config);
+            check.CheckString(serialized, checkName, fileExtension, memberName, sourceFilePath);
+        }
+
+        [TestMethod]
+        public void SerializeDefaultConfig()
+        {
+            checkConfig(DotvvmConfiguration.CreateDefault());
+        }
+
+        [TestMethod]
+        public void SerializeEmptyConfig()
+        {
+            checkConfig(new DotvvmConfiguration());
+        }
+
+        [TestMethod]
+        public void SerializeRoutes()
+        {
+            var c = new DotvvmConfiguration();
+
+            c.RouteTable.Add("route1", "url1", "file1.dothtml", new { a = "ccc" });
+            c.RouteTable.Add("route2", "url2/{int}", "file1.dothtml", new { a = "ccc" });
+            c.RouteTable.Add("custom presenter", "url3", "", presenterFactory: LocalizablePresenter.BasedOnQuery("lang"));
+
+            c.RouteTable.AddGroup("group1", "group-{lang}/", "", rg => {
+
+                rg.Add("r1", "r1", "g-r1.dothtml");
+
+                rg.AddGroup("g", "g/", "", rg => {
```

Comment:
This is a compilation error. `rg.AddGroup("g", "g/", "", rg => {` there is declaration of existing variable rg. @exyi

Reply:
This should be valid in modern C#

Reply:
I see. I don't know why but my VS was not able to build it. Now it works.